lib/models/builder.js

```diff
@@ -38,23 +38,37 @@ class Builder extends CoreObject {
   }
 
   /**
+   * Loads the build file
+   *
    * @private
    * @method readBuildFile
-   * @param path The file path to read the build file from
    */
-  readBuildFile(path) {
-    // Load the build file
-    path = this.buildFilePath || path;
-
+  readBuildFile() {
     let project = this.project;
     let projectRoot = project.root;
 
-    let buildFile = findBuildFile('ember-cli-build.js', { projectRoot, path });
-    if (buildFile) {
-      return buildFile({ project });
+    let defaultBuildFilePath = project.isEmberCLIAddon() ? 'tests/dummy/ember-cli-build.js' : 'ember-cli-build.js';
+    let buildFilePath = path.resolve(projectRoot, this.buildFilePath || defaultBuildFilePath);
+    // backward compatibility for addons with `ember-cli-build.js` at project root.
+    if (!fs.existsSync(buildFilePath)) {
+      buildFilePath = findUp.sync('ember-cli-build.js', { cwd: projectRoot });
+    }
```

Comment:
Let's simplify, and ensure that the "magic" defaulting / find-up behavior only happens when a `--build-file-path` was **not** passed.

Reply:
Something like:

```js
let { buildFilePath } = this;
if (!buildFilePath) {
  let defaultBuildFilePath = project.isEmberCLIAddon()
    ? 'tests/dummy/ember-cli-build.js'
    : 'ember-cli-build.js';
  buildFilePath = path.join(project.root, defaultBuildFilePath);
  
  if (!fs.existsSync(buildFilePath)) {
    // backward compatibility for addons with `ember-cli-build.js` at project root.
    buildFilePath = findUp.sync('ember-cli-build.js', { cwd: project.root });
  }
}
```